test/Microsoft.Extensions.Logging.Test/LoggerExtensionsTest.cs

```diff
@@ -40,46 +41,48 @@ public void MessageOnly_LogsCorrectValues()
             logger.LogDebug(_state);
             logger.Log(LogLevel.Information, _state);
 
+            var writeList = sink.Writes.ToList();
```

Comment:
Do you expect tests elsewhere to be written like this? Or should they be popping items off the front of the list?

Reply:
Only if you need to check the order. I see most of the tests use something like Assert.Empty(sink.Writes), Assert.Single(sink.Writes, predicate) or Assert.Contains(sink.Writes, w => predicate) which works with just the enumerable interface. Enumerating as a list is the simplest for test back compat for now but using the TryTake method could also work.

Reply:
This test should reflect our recommendations for other repos.

Reply:
I liked the TryTake version of this, much easier to maintain.